src/port/port.tcl

```diff
@@ -3927,6 +3927,161 @@ proc action_echo { action portlist opts } {
     return 0
 }
 
+proc offer_clone {} {
+    global macports::os_platform macports::os_major macports::os_arch \
+        macports::sources macports::autoconf::rsync_path \
+        macports:rsync_options macports::getindex macports::sources_default \
+        macports::sudo_user macports::prefix macports::portdbpath
+
+    # CHECK IF SOURCES.CONF IS STILL DEFAULT (INDICATING THAT USER IS FIRST TIME CONTRIBUTOR)
+    set confpath "${macports::prefix}/etc/macports/sources.conf"
+    if {[catch {exec [macports::findBinary diff] "-q" $confpath "$confpath.default"} _]} {
```

Comment:
We offer this change if `sources.conf` is identical with `sources.conf.default` or this proc has ran before without unexpectedly failing. We keep track that by creating a file `${macports::portdbpath}/contrib`